### PR TITLE
Remove outdated requirement of CCR

### PR DESCRIPTION
**Pull request by dnhatn**
With retention leases, users do not need to set `index.soft_deletes.retention.operations`. This change removes it from the requirements of CCR.

### CODE REVIEW

**Review by jasontedor — APPROVED** [review]
LGTM.

File: docs/reference/ccr/requirements.asciidoc [hunk 1]
```diff
@@ -41,58 +41,3 @@ to a follower the following process will fail due to incomplete history on the l
 The default value is `12h`.
 
 For more information about index settings, see {ref}/index-modules.html[Index modules].
-
-
-[[ccr-overview-beats]]
```
Comment (jrodewig) [inline docs/reference/ccr/requirements.asciidoc]: It looks like the Beats and APM docs link to this section. Can we add a redirect to [`redirects.asciidoc`](https://github.com/elastic/elasticsearch/blob/master/docs/reference/redirects.asciidoc) so these links don't break?

Comment (jasontedor) [reply docs/reference/ccr/requirements.asciidoc]: That note (elastic/beats#9427) can be removed from the docs, as it no longer applies.

Comment (jrodewig) [reply docs/reference/ccr/requirements.asciidoc]: I opened elastic/beats#15506 to remove the note in the Beats/APM docs. The docs CI tests will continue to fail for this PR until that's merged or we add a redirect.

**Comment by dnhatn** [comment]
run elasticsearch-ci/docs

File: docs/reference/ccr/requirements.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/ccr/requirements.asciidoc]: @dnhatn  elastic/beats#15506 is now merged. If you update this branch, the docs CI test should pick up the changes in Beats and pass.

**Review by jrodewig — APPROVED** [review]

**Comment by jrodewig** [comment]
@dnhatn Looks like my PR missed a link. Opened elastic/apm-server#3147 to remove.

**Comment by jrodewig** [comment]
run elasticsearch-ci/docs

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by dnhatn** [comment]
@jasontedor @jrodewig Thanks for reviewing.

**Pull request merged**